moshi/moshi/models/tts.py

```diff
@@ -0,0 +1,624 @@
+        word: word as string.
+        padding: if > 0, we will force that many padding after the word.
+            To have an effect this should be more than len(tokens), as we will
+            prevent new word until we have passed all the text tokens.
+        audio_tokens: is used when some audio should be used as a prefix in the model."""
+    tokens: list[int]
+    word: str
+    padding: int = 0
+    audio_tokens: torch.Tensor | None = None
+
+
+@dataclass
+class State:
+    """State of the TTS Machine.
+
+    Args:
+        entries: queue containing the entries to generate.
+        remaining_padding: how many times the model can still sample a pad.
+        forced_padding: how many times the model is still forced to sample a pad.
+        queued: queue containing the main stream text tokens to feed.
+        lookahead: queue containing the lookahead text tokens to feed.
+        end_step: once we reach the end of the generation, this is set to the current step.
+            The end of the generation is once the model samples a `word` but `entries` is empty.
+        consumption_times: list of steps at which each entry in `entries` was consumed.
+        transcript: list of tuples `(word, step)`, at which each word was consumed.
+        audio_tokens_remaining: when using some audio as prefix, this would contain the remaining
+            audio tokens to force into the model.
+        zero_text_remaining: when using an audio only prefix, for how long we should still force
+            the text to be `zero`.
+    """
+    entries: deque[Entry]
+    remaining_padding: int
+    forced_padding: int
+    queued: deque[int] = field(default_factory=deque)
+    lookahead: deque[int] = field(default_factory=deque)
+    end_step: int | None = None
+    consumption_times: list[int] = field(default_factory=list)
+    transcript: list[tuple[str, int]] = field(default_factory=list)
+
+    def get_tokens_ahead(self, lookahead: int) -> list[int]:
+        assert lookahead > 0
+        for entry in self.entries:
+            if entry.tokens:
+                lookahead -= 1
+                if lookahead == 0:
+                    return entry.tokens
+        return []
+
+
+def _delayed(codes: torch.Tensor, delays: list[int], fill_value: int) -> torch.Tensor:
+    # Apply the acoustic delay on the provided audio tokens.
+    K, T = codes.shape
+    out = torch.full((K, T + max(delays)), fill_value, device=codes.device, dtype=torch.long)
+    for k in range(K):
+        delay = delays[k]
+        out[k, delay: delay + T] = codes[k]
+    return out
+
+
+def _make_null(all_attributes: tp.Sequence[ConditionAttributes]) -> list[ConditionAttributes]:
+    # When using CFG, returns the null conditions.
+    return dropout_all_conditions(all_attributes)
+
+
+@dataclass
+class StateMachine:
+    """State machine that manipulates the `State` based on the model prediction.
+    In particular, every time the model predicts a `word` (see `TokenIds`) special token,
```

Comment:
Cosmetic but maybe it would be clearer to call the `word` token something like `new_word` (as word token seems a bit like a "regular" token).